### PR TITLE
Add carbon theme

### LINKED ISSUE #13673: [FEATURE] Add Carbon Theme
### Problem description

I'd like to create prettier dark plots like
https://medium.com/@pYdeas/create-publication-styled-figures-with-hvplot-using-the-bokeh-backend-b623ab48a5d2

### Feature description

Just an additional theme

### Potential alternatives

Use other themes

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ahuang11**
image

Based on https://medium.com/@pYdeas/create-publication-styled-figures-with-hvplot-using-the-bokeh-backend-b623ab48a5d2

- [x] issues: fixes #13673
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

Tested thru holoviews:
```python
import pandas as pd
import hvplot.pandas
import holoviews as hv
from bokeh.themes import Theme

json = {
    "attrs": {
        "figure": {
            "background_fill_color": "#1b1e23",
            "border_fill_color": "#1b1e23",
            "outline_line_alpha": 0,
        },
        "Grid": {
            "grid_line_color": "#808080",
            "grid_line_alpha": 0.1,
        },
        "Axis": {
            # tick color and alpha
            "major_tick_line_color": "#4d4f51",
            "minor_tick_line_alpha": 0,
            # tick labels
            "major_label_text_font": "Courier New",
            "major_label_text_color": "#808080",
            "major_label_text_align": "left",
            "major_label_text_font_size": "0.95em",
            "major_label_text_font_style": "normal",
            # axis labels
            "axis_label_text_font": "Courier New",
            "axis_label_text_font_style": "normal",
            "axis_label_text_font_size": "1.15em",
            "axis_label_text_color": "lightgrey",
            "axis_line_color": "#4d4f51",
        },
        "Legend": {
            "spacing": 8,
            "glyph_width": 15,
            "label_standoff": 8,
            "label_text_color": "#808080",
            "label_text_font": "Courier New",
            "label_text_font_size": "0.95em",
            "label_text_font_style": "bold",
            "border_line_alpha": 0,
            "background_fill_alpha": 0,
            "inactive_fill_alpha": 0,
        },
        "BaseColorBar": {
            # axis labels
            "title_text_color": "lightgrey",
            "title_text_font": "Courier New",
            "title_text_font_size": "0.95em",
            "title_text_font_style": "normal",
            # tick labels
            "major_label_text_color": "#808080",
            "major_label_text_font": "Courier New",
            "major_label_text_font_size": "0.95em",
            "major_label_text_font_style": "normal",
            "background_fill_color": "#1b1e23",
            "major_tick_line_alpha": 0,
            "bar_line_alpha": 0,
        },
        "Title": {
            "text_font": "Courier New",
            "text_font_style": "normal",
            "text_color": "lightgrey",
        },
    }
}

theme = Theme(json=json)

hv.renderer("bokeh").theme = theme

# load in data and postprocess input
df = (
    pd.read_csv(
        "https://mesonet.agron.iastate.edu/cgi-bin/request/daily.py?network=WA_ASOS&stations=SEA&year1=1928&month1=1&day1=1&year2=2023&month2=12&day2=31&var=max_temp_f&var=min_temp_f&var=precip_in&na=blank&format=csv",
        parse_dates=True,
        index_col="day",
    )
    .drop(columns=["station"])
    .astype("float16")
    .assign(
        dayofyear=lambda df: df.index.dayofyear,
        year=lambda df: df.index.year,
    )
)

# get average and 2023
df_avg = df.loc[df["year"].between(1990, 2020)].groupby("dayofyear").mean()
df_2023 = df[df.year == 2023]

# preprocess below/above
df_above = df_2023[["dayofyear", "max_temp_f"]].merge(
    df_avg.reset_index()[["dayofyear", "max_temp_f"]],
    on="dayofyear",
    suffixes=("_2023", "_avg"),
)
df_above["max_temp_f"] = df_above["max_temp_f_avg"]
df_above["max_temp_f"] = df_above.loc[
    df_above["max_temp_f_2023"] >= df_above["max_temp_f_avg"], "max_temp_f_2023"
]

df_below = df_2023[["dayofyear", "max_temp_f"]].merge(
    df_avg.reset_index()[["dayofyear", "max_temp_f"]],
    on="dayofyear",
    suffixes=("_2023", "_avg"),
)
df_below["max_temp_f"] = df_below["max_temp_f_avg"]
df_below["max_temp_f"] = df_below.loc[
    df_below["max_temp_f_2023"] < df_below["max_temp_f_avg"], "max_temp_f_2023"
]

days_above = df_above.query("max_temp_f_2023 >= max_temp_f_avg")["max_temp_f"].size
days_below = df_below.query("max_temp_f_2023 < max_temp_f_avg")["max_temp_f"].size

# create plot elements
plot = df.hvplot(
    x="dayofyear",
    y="max_temp_f",
    by="year",
    color="grey",
    alpha=0.02,
    legend=False,
    hover=False,
)
plot_avg = df_avg.hvplot(x="dayofyear", y="max_temp_f", color="grey", legend=False)
plot_2023 = df_2023.hvplot(x="dayofyear", y="max_temp_f", color="black", legend=False)
scatter_2023 = df_2023.hvplot.scatter(
    x="dayofyear",
    y="max_temp_f",
    color="max_temp_f",
    size=5,
    legend=False,
    cmap="RdBu_r",
    alpha=0.25,
    clabel="TEMPERATURE °F",
)

dark_red = "#FF5555"
dark_blue = "#5588FF"

plot_above = df_above.hvplot.area(
    x="dayofyear", y="max_temp_f_avg", y2="max_temp_f", label="above", legend="top_right"
).opts(fill_alpha=0.2, line_alpha=0.8, line_color=dark_red, fill_color=dark_red)
plot_below = df_below.hvplot.area(
    x="dayofyear", y="max_temp_f_avg", y2="max_temp_f", label="below", legend="top_right"
).opts(fill_alpha=0.2, line_alpha=0.8, line_color=dark_blue, fill_color=dark_blue)

text_days_above = hv.Text(
    35, df_2023["max_temp_f"].max(), f"{days_above}", fontsize=14
).opts(text_align="right", text_baseline="bottom", text_color=dark_red, text_alpha=0.8)
text_days_below = hv.Text(
    35, df_2023["max_temp_f"].max(), f"{days_below}", fontsize=14
).opts(text_align="right", text_baseline="top", text_color=dark_blue, text_alpha=0.8)
text_above = hv.Text(38, df_2023["max_temp_f"].max(), "DAYS ABOVE", fontsize=7).opts(
    text_align="left", text_baseline="bottom", text_color="lightgrey", text_alpha=0.8
)
text_below = hv.Text(38, df_2023["max_temp_f"].max(), "DAYS BELOW", fontsize=7).opts(
    text_align="left", text_baseline="above", text_color="lightgrey", text_alpha=0.8
)

# overlay every ing and save
final = (
    plot
    * plot_avg
    * plot_above
    * plot_below
    * scatter_2023
    * text_days_above
    * text_days_below
    * text_above
    * text_below
).opts(
    xlabel="TIME OF YEAR",
    ylabel="MAX TEMP °F",
    title="SEATTLE 2023 vs AVERAGE (1990-2020)",
    gridstyle={"ygrid_line_alpha": 0},
    xticks=[
        (1, "JAN"),
        (31, "FEB"),
        (59, "MAR"),
        (90, "APR"),
        (120, "MAY"),
        (151, "JUN"),
        (181, "JUL"),
        (212, "AUG"),
        (243, "SEP"),
        (273, "OCT"),
        (304, "NOV"),
        (334, "DEC"),
    ],
    show_grid=True,
    fontscale=1.18,
)
hvplot.show(final)

```

### CODE REVIEW

File: src/bokeh/themes/__init__.py [hunk 1]
```diff
@@ -92,6 +99,7 @@
 
 built_in_themes = {
     CALIBER       : Theme(json=_caliber.json),
+    CARBON        : Theme(json=_caliber.json),
```
Comment (ahuang11) [inline src/bokeh/themes/__init__.py]:
```suggestion
    CARBON        : Theme(json=_carbon.json),
```

File: src/bokeh/themes/__init__.py [hunk 2]
```diff
@@ -92,8 +100,9 @@
 
 built_in_themes = {
     CALIBER       : Theme(json=_caliber.json),
+    CARBON        : Theme(json=_carbon.json),
     DARK_MINIMAL  : Theme(json=_dark_minimal.json),
-    LIGHT_MINIMAL : Theme(json=_caliber.json),
+    LIGHT_MINIMAL : Theme(json=_light_minimal.json),
```
Comment (mosc9575) [inline src/bokeh/themes/__init__.py]: Please change the target branch to `branch-3.4`. The branch `bokeh:main` you selected was updated for the last time two years ago and this line was already corrected in this [commit](https://github.com/bokeh/bokeh/commit/c9c0adfae166d7efd1af3b08143af0e6fe449812#diff-c4ad4f86d32a852894df761b1ee4579d61d6b2e4ee43485a6e7729ad17c6a90e).

Please also add a screenshot of your example to show how it looks like.

**Comment by mattpap** [comment]
@ahuang11, this needs to be rebased onto `branch-3.4`.

**Comment by MarcSkovMadsen** [comment]
Hi @ahuang11 . This looks awesome. Have you tested on a wide fangevogtere plots?

In my experience a theme like that needs a lot work to work well across plot types. For example the theme we are using with the Panel FastTemplate looks unfinished for violinplots.

**Comment by ahuang11** [comment]
Created another PR that's rebased https://github.com/bokeh/bokeh/pull/13675

**Comment by mattpap** [comment]
> Created another PR that's rebased 

For the future, we could prefer work to continue in an existing PR. For that a branch can be rebased with `git rebase --onto origin/branch-3.4 INITIAL_SHA_OF_A_BRANCH~1`.

**Comment by ahuang11** [comment]
Since the branch was so outdated I was worried that there would be many rebase conflicts, but if not, good to to know for the future.

**Comment by mattpap** [comment]
> Since the branch was so outdated I was worried that there would be many rebase conflicts, but if not, good to to know for the future.

This is the function of `--onto BRANCH_NAME` argument to `git rebase`, which approximately does a cherry pick of a commit range onto a new base, thus there would either be no conflicts or only minor conflicts to deal with. If it was easier for you to start over in a new PR, that's fine, but it's usually more work than doing an onto rebase.

**Pull request closed without merging**